### PR TITLE
Re-enable mdx tests

### PR DESCRIPTION
**Pull request by NathanReb**
We recently released `mdx.1.5.0` which contains fixes for some of the bugs you ran into so I took the liberty to open a PR re-enabling the mdx check of the README.

I also rewrote the dune rules by hand as `mdx rule` still needs to be improved a little. As a matter of fact it's going to be replaced by a proper dune integration for mdx in an hopefully near future!

Let me know what you think!

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -1 +1 @@
-(lang dune 1.0)
```
Comment (c-cube) [inline dune-project]: why is that needed? :)

Comment (NathanReb) [reply dune-project]: The new rules I wrote use some `dune.1.11` features such as the `target` field. I can try to lower the required version down a bit if you want!

Comment (c-cube) [reply dune-project]: were the old rules wrong?

Comment (NathanReb) [reply dune-project]: Sorry, I forgot to answer! Yeah using a `progn` in this case isn't very friendly to dune as it doesn't know about generated files. Also the `diff?` behaviour isn't the best, we've had issue with these mdx rules in the past.

File: dune [hunk 2]
```diff
@@ -1,8 +1,8 @@
+(rule
+ (target README.md.corrected)
+ (deps (package containers))
+ (action (run ocaml-mdx test %{dep:README.md} -o %{target})))
```
Comment (c-cube) [inline dune]: btw @NathanReb I get messages like that:

```
ocaml-mdx: unknown option `-o'.
Usage: ocaml-mdx test [OPTION]... FILE
Try `ocaml-mdx test --help' or `ocaml-mdx --help' for more information.
``` and I'm not sure why?

Comment (NathanReb) [reply dune]: You need to upgrade to `mdx.1.5.0`! The `-o` option was introduced in that version.
